### PR TITLE
Feature/four 19267

### PR DESCRIPTION
**Pull request by luNunezProcessmaker**
## Issue & Reproduction Steps

- Return the column required
- Enable the sort by columns
- Consider the page filters

like:
```
$request->validate([
           'case_number' => 'required|integer',
            'status' => 'nullable|string|in:ACTIVE,INACTIVE',
            'per_page' => 'nullable|integer',
            'per_page' => 'nullable|integer',
            'order_by' => 'nullable|string|in:id,column1, column2',
            'order_direction' => 'nullable|string|in:asc,desc',
```

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-19267

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:deploy
ci:next

### CODE REVIEW

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-5239538ebd.engk8s.processmaker.net

File: ProcessMaker/Http/Controllers/Api/ProcessRequestController.php [hunk 1]
```diff
@@ -839,4 +839,44 @@ public function endEventDestination(ProcessRequest $request)
...
+        $case_number = $request->input('case_number', 0);
+        $query = ProcessRequest::forUser($user);
+        $includes = $request->input('include', '');
+        foreach (array_filter(explode(',', $includes)) as $include) {
+            if (in_array($include, ProcessRequest::$allowedIncludes)) {
+                $query->with($include);
+            }
+        }
+
+        if ($advancedFilter = $request->input('advanced_filter', '')) {
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/ProcessRequestController.php:863]: What is the purpose of this advanced filter?

File: ProcessMaker/Http/Controllers/Api/ProcessRequestController.php [hunk 2]
```diff
@@ -839,4 +839,44 @@ public function endEventDestination(ProcessRequest $request)
...
+            Filter::filter($query, $advancedFilter);
+        }
+
+        $response = $query->where('case_number', $case_number)
+            ->orderBy(
+                str_ireplace('.', '->', $request->input('order_by', 'name')),
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/ProcessRequestController.php:869]: Why you are using this str_ireplace? what is the use case to tackle this?

File: ProcessMaker/Http/Controllers/Api/ProcessRequestController.php [hunk 3]
```diff
@@ -839,4 +839,44 @@ public function endEventDestination(ProcessRequest $request)
...
+                $request->input('order_direction', 'ASC')
+            )
+            ->select('process_requests.*')
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/ProcessRequestController.php:872]: Avoid to select *, only use the necesary columns here

File: ProcessMaker/Http/Controllers/Api/ProcessRequestController.php [hunk 4]
```diff
@@ -839,4 +839,44 @@ public function endEventDestination(ProcessRequest $request)
...
+            ->withAggregate('processVersion', 'alternative')
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/ProcessRequestController.php:873]: Why you are added this alternative?

File: ProcessMaker/Http/Controllers/Api/ProcessRequestController.php [hunk 5]
```diff
@@ -839,4 +839,44 @@ public function endEventDestination(ProcessRequest $request)
...
+     *  
+     * @param Request $request
+     *
+     * @return ApiCollection
+     */
+    public function getRequestsByCase(Request $request)
+    {
+        $user = Auth::user();
+
+        $case_number = $request->input('case_number', 0);
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/ProcessRequestController.php:854]: Please validate the parameters send

File: tests/Feature/Api/ProcessRequestsTest.php [hunk 6]
```diff
@@ -986,4 +986,32 @@ public function testScreenRequested()
         $data = $response->json()['data'];
         $this->assertEmpty($data);
     }
+    
+    /**
+     * Get a list of Requests by Cases.
+     */
+    public function testRequestByCase()
```
Comment (pmPaulis) [inline tests/Feature/Api/ProcessRequestsTest.php:993]: Add a test what happen if the case_number was not send???

**Review by pmPaulis — CHANGES_REQUESTED** [review]

File: ProcessMaker/Http/Controllers/Api/ProcessRequestController.php (hunk 1 above)
Comment (luNunezProcessmaker) [reply ProcessMaker/Http/Controllers/Api/ProcessRequestController.php:863]: this is used, when is filter by status column

File: ProcessMaker/Models/ProcessRequest.php [hunk 7]
```diff
@@ -1089,4 +1089,36 @@ public function getElementDestination(): ?array
...
+    public function scopeApplyPagination($query, $request)
+    {
+        $page = $request->input('page', 1);
+        $perPage = $request->input('per_page', 10);
+
+        return $query->paginate($perPage);
+    }
+
+    /**
+     * Scope to filter by case_number through the processRequest relationship
```
Comment (pmPaulis) [inline ProcessMaker/Models/ProcessRequest.php:1116]: remove this `through the processRequest relationship`

File: tests/Feature/Api/ProcessRequestsTest.php [hunk 8]
```diff
@@ -986,4 +986,46 @@ public function testScreenRequested()
...
+    {
+        ProcessRequest::query()->delete();
+        $request = ProcessRequest::factory()->create();
+        ProcessRequest::factory()->count(9)->create([
+            'parent_request_id' => $request->id,
+        ]);
+
+        $url = '/requests-by-case?case_number=' . $request->case_number;
```
Comment (pmPaulis) [inline tests/Feature/Api/ProcessRequestsTest.php:1001]: You can define in a constant the URI to use like:
`const API_REQUESTS_BY_CASE = '/requests-by-case';`

**Review by pmPaulis — CHANGES_REQUESTED** [review]

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-5239538ebd.engk8s.processmaker.net

File: ProcessMaker/Http/Controllers/Api/ProcessRequestController.php [hunk 9]
```diff
@@ -839,4 +839,44 @@ public function endEventDestination(ProcessRequest $request)
...
+    public function getRequestsByCase(Request $request, User $user = null)
+    {
+        if (!$user) {
+            $user = Auth::user();
+        }
+
+        // Validate the inputs, including optional ones
+        $request->validate([
+            'case_number' => 'required|integer',
+            'status' => 'nullable|string|in:ACTIVE,COMPLETED,ERROR,CANCELED',
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/ProcessRequestController.php:859]: Maybe ERROR is not necesary

File: tests/Feature/Api/ProcessRequestsTest.php [hunk 10]
```diff
@@ -986,4 +986,46 @@ public function testScreenRequested()
         $data = $response->json()['data'];
         $this->assertEmpty($data);
     }
+    
+    /**
+     * Get a list of Requests by Cases.
+     */
+    public function testRequestByCase()
```
Comment (pmPaulis) [inline tests/Feature/Api/ProcessRequestsTest.php:993]: Please create a test to get by status 'COMPLETED'

**Review by pmPaulis — CHANGES_REQUESTED** [review]

File: ProcessMaker/Http/Controllers/Api/ProcessRequestController.php [hunk 11]
```diff
@@ -839,4 +839,44 @@ public function endEventDestination(ProcessRequest $request)
...
+    public function getRequestsByCase(Request $request, User $user = null)
+    {
+        if (!$user) {
+            $user = Auth::user();
+        }
+
+        // Validate the inputs, including optional ones
+        $request->validate([
+            'case_number' => 'required|integer',
+            'status' => 'nullable|string|in:ACTIVE,COMPLETED,CANCELED',
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/ProcessRequestController.php:859]: remove the status

**Review by pmPaulis — CHANGES_REQUESTED** [review]

**Review by pmPaulis — APPROVED** [review]

**Pull request merged**